frontend/src/lib/components/CommandPalette/DebugCHQueries.tsx

```diff
@@ -1,129 +1,158 @@
+    const { setPathFilter, loadQueries } = useActions(debugCHQueriesLogic)
 
-    const has5lines = nthChar(item.query, '\n', 5)
-    if (has5lines === -1) {
-        return (
-            <CodeSnippet language={Language.SQL} copyDescription="Copy query" style={{ maxWidth: 600, fontSize: 12 }}>
-                {item.query}
-            </CodeSnippet>
-        )
-    }
     return (
         <>
-            <CodeSnippet
-                hideCopyButton={!expanded}
-                language={Language.SQL}
-                copyDescription="Copy query"
-                style={{ maxWidth: 600, fontSize: 12 }}
-            >
-                {expanded ? item.query : item.query.slice(0, has5lines)}
-            </CodeSnippet>
-            <LemonButton size="small" onClick={() => setExpanded(!expanded)}>
-                {expanded ? 'Show less' : 'Show more'}
-            </LemonButton>
-        </>
-    )
-}
-
-function ModalContent({ origResult }: { origResult: Query[] }): JSX.Element {
-    const [pathFilter, setPathFilter] = useState(null as string | null)
-
-    const paths = Object.entries(
-        origResult
-            .map((result) => result.path)
-            .reduce((acc, val) => {
-                acc[val] = acc[val] === undefined ? 1 : (acc[val] += 1)
-                return acc
-            }, {})
-    ).sort((a: any, b: any) => b[1] - a[1])
-
-    const results = pathFilter ? origResult.filter((item) => item.path === pathFilter) : origResult
-
-    return (
-        <>
-            {paths.map(([path, count]) => (
-                <LemonButton
-                    key={path}
-                    type={pathFilter === path ? 'primary' : 'tertiary'}
-                    size="small"
-                    onClick={() => (pathFilter === path ? setPathFilter(null) : setPathFilter(path))}
-                >
-                    {path} ({count})
-                </LemonButton>
-            ))}
-            {pathFilter && <LemonButton onClick={() => setPathFilter(null)}>Remove path filter</LemonButton>}
+            {!!paths?.length && (
+                <div className="flex gap-4 items-end justify-between mb-4">
+                    <div className="flex flex-wrap gap-2">
+                        {paths.map(([path, count]) => (
+                            <LemonButton
+                                key={path}
+                                type={pathFilter === path ? 'primary' : 'tertiary'}
+                                size="small"
+                                onClick={() => (pathFilter === path ? setPathFilter(null) : setPathFilter(path))}
+                            >
+                                {path} ({count})
+                            </LemonButton>
+                        ))}
+                    </div>
+                    <LemonButton icon={<IconRefresh />} onClick={() => loadQueries()} size="small" type="secondary">
+                        Refresh
+                    </LemonButton>
```

Comment:
nit: adding a loading state here
```suggestion
                    <LemonButton
                        icon={<IconRefresh />}
                        loading={queriesLoading}
                        onClick={() => loadQueries()}
                        size="small"
                        type="secondary"
                    >
                        Refresh
                    </LemonButton>
```

Reply:
Hmm, in that case it would be redundant to have a loading state, because the primary element of the modal – the table – already does this. And usually showing the loading state for a single piece of data in multiple places at once doesn't look ideal. But this button should definitely be disabled during loading.